src/urllib3/connection.py

```diff
@@ -80,31 +108,57 @@ class HTTPConnection(_HTTPConnection):
       Or you may want to disable the defaults by passing an empty list (e.g., ``[]``).
     """
 
-    default_port = port_by_scheme["http"]
-
-    #: Disable Nagle's algorithm by default.
-    #: ``[(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)]``
-    default_socket_options = [(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)]
+    default_port: int = port_by_scheme["http"]
 
     #: Whether this connection verifies the host's certificate.
-    is_verified = False
+    is_verified: bool = False
+
+    source_address: Optional[Tuple[str, int]]
+    socket_options: connection.SocketOptions
+    _tunnel_host: Optional[str]
+    _tunnel: Callable[["HTTPConnection"], None]
 
-    def __init__(self, *args, **kw):
+    def __init__(
+        self,
+        host: str,
+        port: Optional[int] = None,
+        timeout: Optional[float] = connection.SOCKET_GLOBAL_DEFAULT_TIMEOUT,
+        source_address: Optional[Tuple[str, int]] = None,
+        blocksize: int = 8192,
+        # Disable Nagle's algorithm by default.
+        socket_options: connection.SocketOptions = default_socket_options,
+        proxy: Optional[str] = None,
+        proxy_config: Optional[ProxyConfig] = None,
+    ) -> None:
         # Pre-set source_address.
-        self.source_address = kw.get("source_address")
+        self.source_address = source_address
 
-        #: The socket options provided by the user. If no options are
-        #: provided, we use the default options.
-        self.socket_options = kw.pop("socket_options", self.default_socket_options)
+        self.socket_options = socket_options
 
         # Proxy options provided by the user.
-        self.proxy = kw.pop("proxy", None)
-        self.proxy_config = kw.pop("proxy_config", None)
-
-        super().__init__(*args, **kw)
+        self.proxy = proxy
+        self.proxy_config = proxy_config
+
+        if sys.version_info >= (3, 7):
+            super().__init__(
+                host=host,
+                port=port,
+                timeout=timeout,
+                source_address=source_address,
+                blocksize=blocksize,
+            )
+        else:
+            super().__init__(
+                host=host, port=port, timeout=timeout, source_address=source_address
+            )
 
-    @property
-    def host(self):
+    # https://github.com/python/mypy/issues/4125
+    # Mypy treats this as LSP violation, which is considered a bug.
+    # If `host` is made a property it violates LSP, because a writeable attribute is overriden with a read-only one.
+    # However, there is also a `host` setter so LSP is not violated.
+    # Potentailly, a `@host.deleter` might be needed depending on how this issue will be fixed.
+    @property  # type: ignore
```

Comment:
I think it is a mypy bug, hence `type: ignore`

Reply:
Has this bug been reported to `mypy`? If not might be something to report and get their thoughts on.

Reply:
It's been an issue for some time now - https://github.com/python/mypy/issues/4125